api/src/main/java/au/org/aodn/nrmn/restapi/validation/validators/global/raw/ATRCSurveyGroupComplete.java

```diff
@@ -30,31 +33,59 @@ public ATRCSurveyGroupComplete() {
+                .collect(Collectors.groupingBy(stagedSurveyMethod -> new Tuple3(stagedSurveyMethod.getSiteCode(),
                         stagedSurveyMethod.getDate(), stagedSurveyMethod.getDepth())));
-        return transectsGroupedBySurveyGroup.entrySet()
-                                     .stream()
-                                     .map((entry) -> validateSurveyGroup(job, entry.getKey(), entry.getValue()))
-                                     .reduce(Validated.valid("survey group is complete: nothing to validate"), (acc,
-                                      validator) ->
-                        acc.combine(Monoids.stringConcat, validator)
-                );
+        return transectsGroupedBySurveyGroup.entrySet().stream()
+                .map((entry) -> validateSurveyGroup(job, entry.getKey(), entry.getValue()))
+                .reduce(Validated.valid("survey group is complete: nothing to validate"),
+                        (acc, validator) -> acc.combine(Monoids.stringConcat, validator));
     }
 
     private Validated<StagedRowError, String> validateSurveyGroup(StagedJob job, Tuple3 surveyGroupKey,
-     List<StagedSurveyTransect> transects) {
-        val surveyNums = transects.stream()
-                              .map(StagedSurveyTransect::getSurveyNum)
-                              .collect(Collectors.toList());
-
-        if (surveyGroupComplete(surveyNums)) {
-            return Validated.valid("surveyGroup " + surveyGroupKey + ": is complete");
-        } else {
-            return invalid(job.getId(),
-                    surveyGroupKey + " has incorrect set of surveyNums: " + surveyNums,
+            List<StagedSurveyTransect> transects) {
+
+        List<String> surveyNums = transects.stream().map(StagedSurveyTransect::getSurveyNum)
+                .collect(Collectors.toList());
+
+        if (!surveyGroupComplete(surveyNums)) {
+            return invalid(job.getId(), surveyGroupKey + " has incorrect set of surveyNums: " + surveyNums,
                     ValidationLevel.BLOCKING);
         }
+
+        val surveys = transects.stream().collect(Collectors.groupingBy(StagedSurveyTransect::getSurveyNum));
+
+        List<Validated<StagedRowError, String>> result = new ArrayList<Validated<StagedRowError, String>>();
+
+        surveys.keySet().stream().forEach(key -> {
+
+            List<StagedSurveyTransect> surveyRecords = surveys.get(key);
+            Map<String, List<StagedSurveyTransect>> surveyByMethod = surveyRecords.stream()
+                    .collect(Collectors.groupingBy(StagedSurveyTransect::getMethod));
+
+            surveyByMethod.keySet().stream().forEach(method -> {
+
+                List<StagedSurveyTransect> rows = surveyByMethod.get(method);
+                String surveyName = rows.get(0).getDepth() + "." + rows.get(0).getSurveyNum();
+                String errorMessage = "Survey " + surveyName + " is missing Method " + method;
```

Comment:
This is probably not enough information for them to be able to find the problem survey - the spreadsheet can contain more than one site/date.

Reply:
Will now display the site code/date/survey at the start of the message.